config/deploy.js

```diff
@@ -129,6 +133,42 @@ function deployBuild(url, folder) {
     }
 }
 
+function buildAssetHistory(url) {
+    shell.exec('git clone ' + url + ' current_build');
+    let now = (new Date().getTime()) / 1000;
+    let oldHistoryPath = path.resolve(__dirname, 'current_build/history.json');
+    let newHistoryPath = path.resolve(__dirname, 'history.json');
+    let oldAssetsPath = path.resolve(__dirname, 'current_build/bundles');
+    let newAssetsPath = path.resolve(__dirname, 'bundles');
+
+    // Parse old history file if it exists
+    let history = {};
+    try {
+        history = JSON.parse(fs.readFileSync(oldHistoryPath).toString());
+    } catch (e) {}
+
+    // Add new asset files to history, with current timestamp
+    let newAssets = fs.readdirSync(newAssetsPath);
+    for (asset of newAssets) {
+        history[asset] = { time: now };
+    }
+
+    try {
+        fs.accessSync(oldAssetsPath);
+        let existingAssets = fs.readdirSync(oldAssetsPath);
+        for (asset of existingAssets) {
+            let assetPath = path.resolve(newAssetsPath, asset);
+            // Check if old assets don't name-conflict and are less than six months old
+            if (!fs.existsSync(assetPath) && (now - history[asset].time < 15768000)) {
```

Comment:
Discussed offline. For right now, have decided to keep the history file for all files, in the event that users start running into script lab errors, we could look up their files here.